src/server/qgsfcgiserverrequest.cpp

```diff
@@ -106,11 +106,31 @@ QgsFcgiServerRequest::QgsFcgiServerRequest()
   setUrl( url );
   setMethod( method );
 
-  // Get accept header for content-type negotiation
-  const char *accept = getenv( "HTTP_ACCEPT" );
-  if ( accept )
+  // Fill the headers dictionary
+  const QStringList headers
   {
-    setHeader( QStringLiteral( "Accept" ), accept );
+    QStringLiteral( "Accept" ),
+    QStringLiteral( "User-Agent" ),
+    QStringLiteral( "Proxy" ),
+    QStringLiteral( "Authorization" ),
+    QStringLiteral( "X-Qgis-Service-Url" ),
+    QStringLiteral( "X-Qgis-WMS-Service-Url" ),
+    QStringLiteral( "X-Qgis-WFS-Service-Url" ),
+    QStringLiteral( "X-Qgis-WCS-Service-Url" ),
+    QStringLiteral( "X-Qgis-WMTS-Service-Url" ),
+    QStringLiteral( "Forwarded" ),
+    QStringLiteral( "X-Forwarded-Host" ),
+    QStringLiteral( "X-Forwarded-Proto" ),
+    QStringLiteral( "Host" )
```

Comment:
What do you think about moving this list in the `QgsServerSettingsEnv::EnvVar` enum with the `HTTP_` prefix instead?

Reply:
or using/completing those defined in the `QgsServerRequest::RequestHeader` enum?

Reply:
@pblottiere what do you think?

Reply:
I'm not sure, I'm going to take a look. And yes I was indeed thinking about relying on a `QMetaEnum`.

Reply:
I'm starting on an implementation based on QMetaEnum

Reply:
Thanks, we don't have redundant code now thanks to your last modifications :+1: